src/components/owned/OwnerInfo.tsx

```diff
@@ -72,39 +70,32 @@ function BadgeTitle({ badge }: { badge: BaseBadgeContract }) {
 }
 
 function BadgeContent({ badge }: { badge: BaseBadgeContract }) {
-  if (badge instanceof EmailBadgeContract) {
+  if (badge.type === 'Email') {
     return (
       <BodyText>
         This is a zkNFT derivative of an email. It means this person has been
         verified own a ‘
-        <AccentText color="text-secondary">{badge.domain}</AccentText>‘ email.
+        <AccentText color="text-secondary">{badge.original}</AccentText>‘ email.
         <ShareCTAButtons />
       </BodyText>
     )
   }
 
-  if (badge instanceof ERC721BadgeContract) {
-    return (
-      <BodyText>
-        This is a zkNFT derivative. It means this person has been verified to
-        own at least one ‘
-        <ExternalLink
-          url={getEtherscanAddressUrl(badge.originalERC721, badge.network)}
-        >
-          <AccentText color="text-secondary">
-            <ContractName
-              address={badge.originalERC721}
-              network={badge.network}
-            />
-          </AccentText>
-        </ExternalLink>
-        ‘ {badge.network[0].toUpperCase() + badge.network.slice(1)} NFT.
-        <ShareCTAButtons />
-      </BodyText>
-    )
-  }
+  const network = data[badge.type].source.network
 
-  return null
+  return (
+    <BodyText>
+      This is a zkNFT derivative. It means this person has been verified to own
+      at least one ‘
+      <ExternalLink url={getEtherscanAddressUrl(badge.original, network)}>
+        <AccentText color="text-secondary">
+          <ContractName address={badge.original} network={network} />
+        </AccentText>
+      </ExternalLink>
+      ‘ {network[0].toUpperCase() + network.slice(1)} NFT.
```

Comment:
Uppercase first letter, let's do a `helper` for this, we may need it in the future